src/main/java/org/jenkinsci/plugins/workflow/steps/ErrorInfoStep.java

```diff
@@ -0,0 +1,193 @@
+         * so we compare by stack trace instead.
+         */
+        private @CheckForNull FlowNode getNode() throws IOException {
+            final Set<String> stackTraces = new HashSet<>();
+            for (Throwable t = error; t != null; t = t.getCause()) {
+                stackTraces.add(Functions.printThrowable(t));
+            }
+            Predicate<FlowNode> threwException = new Predicate<FlowNode>() {
+                @Override
+                public boolean apply(FlowNode input) {
+                    if (input instanceof BlockEndNode) {
+                        return false;
+                    }
+                    ErrorAction a = input.getAction(ErrorAction.class);
+                    return (a != null && stackTraces.contains(Functions.printThrowable(a.getError())));
+                }
+            };
+            return new FlowScanner.DepthFirstScanner().findFirstMatch(getExecution(), threwException);
+        }
+
+        @Whitelisted
+        public @Nonnull Throwable getError() {
+            return error;
+        }
+
+        /**
+         * Gets the stack trace of the error, or just the message in the case of {@link AbortException}.
+         */
+        @Whitelisted
+        public @Nonnull String getStackTrace() {
+            if (error instanceof AbortException) {
+                return error.getMessage();
+            } else {
+                return Functions.printThrowable(error);
+            }
+        }
+
+        /**
+         * Gets the {@link Result} of the build if the error were uncaught.
+         * @return typically {@link Result#FAILURE} but {@link FlowInterruptedException} may override
+         */
+        @Whitelisted
+        public @Nonnull String getResult() {
+            Result r;
+            if (error instanceof FlowInterruptedException) {
+                r = ((FlowInterruptedException) error).getResult();
+            } else {
+                r = Result.FAILURE;
+            }
+            return r.toString();
+        }
+
+        /**
+         * Looks for the URL of the {@link LogAction} last printed before the node which broke.
+         */
+        @Whitelisted
+        public @CheckForNull String getLogURL() throws IOException {
+            FlowNode n = getNode();
+            if (n != null) {
+                FlowNode logNode = new FlowScanner.LinearBlockHoppingScanner()
+                        .findFirstMatch(n,
+                                FlowScanner.MATCH_HAS_LOG);
+                if (logNode != null) {
+                    String u = Jenkins.getActiveInstance().getRootUrl();
+                    if (u == null) {
+                        u = "http://jenkins/"; // placeholder
+                    }
+                    return u + logNode.getUrl() + logNode.getAction(LogAction.class).getUrlName();
+                }
+            }
+            return null;
+        }
+
+        // TODO tail of log
+        // TODO label
```

Comment:
So what would this look like? (Searching for `LabelAction` and/or `StageAction` as per `FlowNodeSerialWalker`’s convenience API for labels.)